src/main/java/ua/javarush/island/entity/Animal.java

```diff
@@ -0,0 +1,136 @@
+package ua.javarush.island.entity;
+
+import ua.javarush.island.gameisland.Area;
+import ua.javarush.island.gameisland.Coordinate;
+import ua.javarush.island.gameisland.Direction;
+import ua.javarush.island.gameisland.Island;
+import java.util.Map;
+import java.util.SplittableRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+
+public abstract class Animal extends Organism {
+
+    private boolean isMoved = false;
+    protected int speedOfMove;
+    protected int foodWeigthForSaturation;
+    protected int currentFoodWeigthForSaturation;
+    protected Map<String, Integer> listOfEatableOrganism;
+
+    public Area getCurrentArea() {
+        return currentArea;
+    }
+
+    public void setCurrentArea(Area currentArea) {
+        this.currentArea = currentArea;
+    }
+
+    public void setSpeedOfMove(int speedOfMove) {
+        this.speedOfMove = speedOfMove;
+    }
+
+    public void setFoodWeigthForSaturation(int foodWeigthForSaturation) {
+        this.foodWeigthForSaturation = foodWeigthForSaturation;
+    }
+
+    public void setListOfEatableOrganism(Map<String, Integer> listOfEatableOrganism) {
+        this.listOfEatableOrganism = listOfEatableOrganism;
+    }
+
+    public void move() {
+
+        // If animal moved in forward area, method move() will call again. As is  result in one iteration
+        // animal can make any times steps. For this reason used isMoved flag.
+
+        if (!isMoved) {
+            SplittableRandom random = new SplittableRandom();
+
+            Coordinate newCoordinate = new Coordinate(0, 0);
+            newCoordinate.setX(currentArea.getCoordinate().getX());
+            newCoordinate.setY(currentArea.getCoordinate().getY());
+
+            Direction direction = choseDirection(random);
+
+            int randomAxisValue = random.nextInt(-speedOfMove, speedOfMove + 1);
+
+            if (direction == Direction.Horizontal) {
+
+                int newAxisValue = newCoordinate.getX() + randomAxisValue;
+                newCoordinate.setX(correctionCoordinate(newAxisValue, direction));
+
+            }
+            if (direction == Direction.Vertical) {
+
+                int newAxisValue = newCoordinate.getY() + randomAxisValue;
+                newCoordinate.setY(correctionCoordinate(newAxisValue, direction));
+            }
+
+            Area desiredArea = Island.areas[newCoordinate.getX()][newCoordinate.getY()];
+
+            tryToMoveAnimalToDesiredArea(desiredArea);
+
+        } else isMoved = false;
+    }
+
+    private void tryToMoveAnimalToDesiredArea(Area desiredArea){
```

Comment:
We need synchronization between 2 areas when we move animals from one directory to another.